candle-transformers/src/models/debertav2.rs

```diff
@@ -0,0 +1,1499 @@
+            query_layer.matmul(&div)?
+        };
+
+        if self.relative_attention {
+            let rel_embeddings = self
+                .pos_dropout
+                .as_ref()
+                .ok_or(candle::Error::Msg(
+                    "relative_attention requires pos_dropout".to_string(),
+                ))?
+                .forward(rel_embeddings)?
+                .unwrap();
+
+            rel_att = Some(self.disentangled_attention_bias(
+                query_layer,
+                key_layer,
+                relative_pos,
+                rel_embeddings,
+                scale_factor,
+            )?);
+        }
+
+        if rel_att.is_some() {
+            attention_scores = attention_scores.broadcast_add(&rel_att.unwrap())?;
+        }
+
+        attention_scores = attention_scores.reshape((
+            (),
+            self.num_attention_heads,
+            attention_scores.dim(D::Minus2)?,
+            attention_scores.dim(D::Minus1)?,
+        ))?;
+
+        let mut attention_probs =
+            XSoftmax::apply(&attention_scores, attention_mask, D::Minus1, &self.device)?;
+
+        attention_probs =
+            self.dropout
+                .forward(Some(&attention_probs))?
+                .ok_or(candle::Error::Msg(
+                    "Dropout did not return a value".to_string(),
+                ))?;
+
+        let mut context_layer = attention_probs
+            .reshape((
+                (),
+                attention_probs.dim(D::Minus2)?,
+                attention_probs.dim(D::Minus1)?,
+            ))?
+            .matmul(&value_layer)?;
+
+        context_layer = context_layer
+            .reshape((
+                (),
+                self.num_attention_heads,
+                context_layer.dim(D::Minus2)?,
+                context_layer.dim(D::Minus1)?,
+            ))?
+            .permute((0, 2, 1, 3))?
+            .contiguous()?;
+
+        let dims = context_layer.dims();
+
+        context_layer = match dims.len() {
+            2 => context_layer.reshape(())?,
+            3 => context_layer.reshape((dims[0], ()))?,
+            4 => context_layer.reshape((dims[0], dims[1], ()))?,
+            5 => context_layer.reshape((dims[0], dims[1], dims[2], ()))?,
+            _ => {
+                return Err(candle::Error::Msg(format!(
+                    "Invalid shape for DisentabgledSelfAttention context layer: {:?}",
+                    dims
+                )))
+            }
+        };
+
+        Ok(context_layer)
+    }
+
+    fn transpose_for_scores(&self, xs: &Tensor) -> candle::Result<Tensor> {
+        let dims = xs.dims().to_vec();
+        let result = match dims.len() {
+            3 => {
+                let reshaped = xs.reshape((dims[0], dims[1], self.num_attention_heads, ()))?;
+
+                let new_dims = reshaped.dims();
+
+                reshaped.transpose(1, 2)?.contiguous()?.reshape((
+                    (),
+                    new_dims[1],
```

Comment:
`new_dims.dim(1)?` make a proper error rather than a panic if there is an issue.